pkg/controller/azvolume.go

```diff
@@ -140,7 +140,26 @@ func (r *reconcileAzVolume) TriggerDelete(ctx context.Context, volumeName string
 	if err := r.UpdateStatus(ctx, azVolume.Name, true); err != nil {
 		return err
 	}
-	klog.Infof("successfully deleted volume (%s)and update status of AzVolume (%s)", azVolume.Spec.UnderlyingVolume, azVolume.Name)
+
+	// Delete all AzVolumeAttachment objects bound to the deleted AzVolume
+	attachments, err := r.azVolumeClient.DiskV1alpha1().AzVolumeAttachments(r.namespace).List(ctx, metav1.ListOptions{})
+	if err != nil && !errors.IsNotFound(err) {
+		klog.Errorf("failed to get AzVolumeAttachments: %v", err)
+		return err
+	}
+
+	klog.V(5).Infof("number of attachments found: %d", len(attachments.Items))
+	for _, attachment := range attachments.Items {
+		klog.Infof("attachment name: %s, volume: %s", attachment.Name, attachment.Spec.UnderlyingVolume)
+		if attachment.Spec.UnderlyingVolume == volumeName {
```

Comment:
Since we're enumerating all of the `AzVolumeAttachment` instances to find the one(s) representing an attachment of this disk, this query would result in high memory usage (and long service times) when there a several thousand instances. Our scale goal includes clusters with several thousand nodes, and even more pods multiplied by disks, so this is a concern. Is it possible to craft a more efficient query for the instances we're interested in?

Reply:
Good point. I will give it some more thoughts on this.